packages-exp/messaging-exp/src/api.ts

```diff
@@ -141,3 +142,22 @@ export function onBackgroundMessage(
   messaging = getModularInstance(messaging);
   return _onBackgroundMessage(messaging as MessagingService, nextOrObserver);
 }
+
+/**
+ * Enables or disables Firebase Cloud Messaging message delivery metrics export to BigQuery. By
+ * default, message delivery metrics are not exported to BigQuery. Use this method to enable or
+ * disable the export at runtime.
+ *
+ * @param messaging - The `FirebaseMessaging` instance.
+ * @param enable - Whether Firebase Cloud Messaging should export message delivery metrics to
+ * BigQuery.
+ *
+ * @public
+ */
+export function setDeliveryMetricsExportedToBigQueryEnabled(
```

Comment:
So that linked flag above is set at app-level. I also see that Android, for example, has [one set for in-app messaging at the messaging service level](https://firebase.google.com/docs/in-app-messaging/modify-message-behavior?platform=ios#enable_opt-out_message_delivery), that I assume falls back to the value of an app-level flag? Should this fall back to the value of the app-level flag if not set explicitly? Also I think messaging on web may need to have its own service-level flag, similar to the one in Android, which stops the use of messaging entirely if the user opts out.